site/src/nav/DownloadButton.jsx

```diff
@@ -42,49 +42,62 @@ function DownloadButton({ mode, zoom, setZoom }) {
     console.log(bounds);
 
     //Send those to the download engine
-    const minxPath = ["bbox", "minx"];
-    const minyPath = ["bbox", "miny"];
-    const maxxPath = ["bbox", "maxx"];
-    const maxyPath = ["bbox", "maxy"];
+    const xmin = ["bbox", "xmin"];
+    const ymin = ["bbox", "ymin"];
+    const xmax = ["bbox", "xmax"];
+    const ymax = ["bbox", "ymax"];
 
     const readOptions = {
       bbox: bbox,
       bboxPaths: {
-        minxPath,
-        minyPath,
-        maxxPath,
-        maxyPath,
+        xmin,
+        ymin,
+        xmax,
+        ymax,
       },
     };
-
-    let parquetDataset = await new ParquetDataset(DownloadCatalog);
-    const wasmTable = await parquetDataset.read(readOptions);
+    let downloadCatalog = getDownloadCatalog(bbox, visibleTypes);
 
     set_panic_hook();
-    //Create a blob
-    // const binaryDataForDownload = writeGeoParquet(wasmTable);
-    const binaryDataForDownload = writeGeoJSON(wasmTable);
 
-    let blerb = new Blob([binaryDataForDownload], {
-      type: "application/octet-stream",
+    // The catalog contains a base path and then a list of types with filenames. 
+    //First, assemble the parquet datasets in parallel.
+   let datasets = downloadCatalog.types.map(type => {
+      return new ParquetDataset(downloadCatalog.basePath, type.files).then((dataset) => {return {type: type.name, parquet: dataset} });
     });
+      
+    Promise.all(datasets)
+    .then((datasets) => {
+      return datasets.map(dataset => dataset.parquet.read(readOptions).then(reader => {return {type: dataset.type, reader:reader}}));
+    }).then((tableReads) => Promise.all(tableReads)
+    .then((wasmTables) => {
+      wasmTables.map(wasmTable => {
+        if (wasmTable?.reader?.numBatches > 0) {
+          const binaryDataForDownload = writeGeoJSON(wasmTable.reader);
+
+          let blerb = new Blob([binaryDataForDownload], {
+            type: "application/octet-stream",
+          });
+    
+          const url = URL.createObjectURL(blerb);
+          var downloadLink = document.createElement("a");
+          downloadLink.href = url;
+  
+          const center = myMap.getCenter();
+          const zoom = myMap.getZoom();
+          downloadLink.download = `overture-${wasmTable.type}-${zoom}-${center.lat}-${center.lng}.geojson`;
+  
+          document.body.appendChild(downloadLink);
+          downloadLink.click();
+          document.body.removeChild(downloadLink);
+        }
+      })
+    }).then(() => {
+      setLoading(false);
+    })
```

Comment:
You're missing a catch statement here. Ideally we would catch any exceptions which bubbled up and then show an alert to the user.

Reply:
Good call, I'll fix this!